test/lib/y2firewall/clients/auto_test.rb

```diff
@@ -33,11 +33,70 @@
     allow(subject).to receive(:importer).and_return(importer)
   end
 
+  describe "#general_summary" do
+    it "creates a text which includes firewalld overview" do
+      expect(firewalld).to receive(:running?).and_return(true)
+      expect(firewalld).to receive(:enabled?).and_return(true)
+      expect(firewalld).to receive(:default_zone).and_return("public")
+
+      summary = subject.send(:general_summary)
+
+      expect(summary).to match(/Running/)
+      expect(summary).to match(/Enabled/)
+      expect(summary).to match(/Default zone/)
+      expect(summary).to match(/Defined zones/)
+    end
+  end
+
+  describe "#zone_summary" do
+    RELATIONS = {
```

Comment:
I would use `let` to define the relations and the zones and a before loop if needed stubs with `allow`, I even would do at just testing the `summary` method allowing to call the originals of `zone_summary` and `general_summary` so you test them through the public method instead of from private calls

Reply:
ok I'll mix all *summary tests together.